Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/certificate/RawCOSEObject.kt

```diff
@@ -0,0 +1,82 @@
+package de.rki.coronawarnapp.vaccination.core.certificate
+
+import com.google.gson.JsonParseException
+import com.google.gson.TypeAdapter
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.ResponseBody
+import okio.ByteString
+import okio.ByteString.Companion.decodeBase64
+import okio.ByteString.Companion.toByteString
+import org.json.JSONObject
+import retrofit2.Converter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
+import javax.inject.Inject
+
+data class RawCOSEObject(
```

Comment:
Wrapper class around a `ByteString` representing raw COSE data.
`typealias` does not give us compiler checks, it's just a visual hint, extension functions are still on the underlying type.
Ideally we would have wanted `value class` here, but those are still experimental in Kotlin.

With the concrete type, we can have Retrofit/Json adapter for serialization.

Reply:
it is stable in 1.5.0, Luckily there is a PR for that https://github.com/corona-warn-app/cwa-app-android/pull/3130
I think you can use it here

Reply:
That's awesome, I'll have a look after 1.5.0 is merged! Thanks!